src/stream/src/executor/sort.rs

```diff
@@ -0,0 +1,506 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::ops::Bound;
+
+use futures::StreamExt;
+use futures_async_stream::try_stream;
+use risingwave_common::array::{Op, Row, StreamChunk};
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::Datum;
+use risingwave_common::util::select_all;
+use risingwave_storage::table::streaming_table::state_table::StateTable;
+use risingwave_storage::StateStore;
+
+use super::error::StreamExecutorError;
+use super::{
+    expect_first_barrier, ActorContextRef, BoxedExecutor, BoxedMessageStream, Executor, Message,
+    PkIndices, Watermark,
+};
+
+/// [`SortBufferKey`] contains a record's timestamp and pk.
+type SortBufferKey = (Datum, Row);
+/// [`SortBufferValue`] contains a record's operation and value, and also a flag indicating whether
+/// the record has been persisted to storage.
+type SortBufferValue = (Op, Row, bool);
```

Comment:
I think it's not trivial to add an `Op` here as we currently have no way to represent this in the storage. As we only support append-only input now, how about removing this `Op`?